src/gwt/src/org/rstudio/studio/client/workbench/views/environment/dataimport/DataImport.java

```diff
@@ -362,20 +378,23 @@ public void onResponseReceived(DataImportPreviewResponse response)
          {
             if (response.getErrorMessage() != null)
             {
-               gridViewer_.setData(null);
+               response.setColumnDefinitions(lastSuccessfulResponse_);
+               setGridViewerData(response);
                progressIndicator_.onError(response.getErrorMessage());
                return;
             }
             
-            gridViewer_.setOption("nullsAsNAs", "true");
+            lastSuccessfulResponse_ = response;
+            
             gridViewer_.setOption("status",
-                  "Previewing first " + toLocaleString(maxRows_) + " entries");
-            gridViewer_.setOption("ordering", "false");
-            gridViewer_.setOption("rowNumbers", "false");
+                  "Previewing first " + toLocaleString(maxRows_) + 
+                  " entries. " + Integer.toString(response.getParsingErrors()) +
```

Comment:
Nit: for simplicity, I'd omit the parsing errors text unless there's actually a parsing error